### PR TITLE
Use shared `leading_quote` implementation in ruff_python_formatter

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

File: crates/ruff_python_stdlib/src/str.rs [hunk 1]
```diff
@@ -11,9 +11,6 @@ pub const SINGLE_QUOTE_PREFIXES: &[&str] = &[
 pub const TRIPLE_QUOTE_SUFFIXES: &[&str] = &["\"\"\"", "'''"];
 pub const SINGLE_QUOTE_SUFFIXES: &[&str] = &["\"", "'"];
 
-pub static STRING_QUOTE_PREFIX_REGEX: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r#"^(?i)[urb]*['"](?P<raw>.*)['"]$"#).unwrap());
```
Comment (charliermarsh) [inline crates/ruff_python_stdlib/src/str.rs:15]: (Drive-by fix -- this is no longer used anywhere, but unrelated to the change at hand.)

Comment (MichaReiser) [reply crates/ruff_python_stdlib/src/str.rs:15]: I noticed that we use `pub` in many places where it isn't necessary. It would be good in my view, to be as restrictive as possible when it comes to defining visibility. Not using `pub` has the advantage that Rust flags unused symbols.

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_ast/src/strings.rs [hunk 2]
```diff
@@ -1,26 +1,33 @@
-use ruff_python_stdlib::str::{
-    SINGLE_QUOTE_PREFIXES, SINGLE_QUOTE_SUFFIXES, TRIPLE_QUOTE_PREFIXES, TRIPLE_QUOTE_SUFFIXES,
-};
-
 /// Strip the leading and trailing quotes from a docstring.
 pub fn raw_contents(contents: &str) -> &str {
-    for pattern in TRIPLE_QUOTE_PREFIXES {
+    for pattern in ruff_python_stdlib::str::TRIPLE_QUOTE_PREFIXES
```
Comment (MichaReiser) [inline crates/ruff_python_ast/src/strings.rs:3]: What's the reason that `raw_contents` is defined in the AST crate vs in the `stdlib` crate? Defining all symbols in the same crates would have the advantage that it may not be necessary for `SINGLE_QUOTE_PREFIXES` to be public. 

Maybe the more fundamental question is what's the difference between the `stdlib` and `ast` crates and is it worth having both or should we merge them into a single `python_syntax` crate? If not, could we add a README to each create (or documentation to `lib.rs`) explaining the intended purpose of each crate.

Comment (charliermarsh) [reply crates/ruff_python_ast/src/strings.rs:3]: I think these should actually be moved into the `ast` crate.

The `stdlib` crate was intended to serve as roughly 1:1 reimplementations of functionality that you typically get from the Python standard library (e.g., lists of keywords, checks for whether something is a valid identifier, lists of standard library modules, and so on).
